comp/networkpath/npcollector/npcollectorimpl/npcollector.go

```diff
@@ -29,9 +67,248 @@ func newNoopNpCollectorImpl() *npCollectorImpl {
+			return
+		case ptest := <-s.pathtestInputChan:
+			s.logger.Debugf("Pathtest received: %+v", ptest)
+			s.receivedPathtestCount.Inc()
+			s.pathtestStore.Add(ptest)
+		}
+	}
+}
+
+func (s *npCollectorImpl) runTracerouteForPath(ptest *pathteststore.PathtestContext) {
+	s.logger.Debugf("Run Traceroute for ptest: %+v", ptest)
+
+	startTime := s.TimeNowFn()
+	cfg := traceroute.Config{
+		DestHostname: ptest.Pathtest.Hostname,
+		DestPort:     ptest.Pathtest.Port,
+		MaxTTL:       0, // TODO: make it configurable, setting 0 to use default value for now
+		TimeoutMs:    0, // TODO: make it configurable, setting 0 to use default value for now
+	}
+
+	path, err := s.runTraceroute(cfg)
+	if err != nil {
+		s.logger.Errorf("%s", err)
+		return
+	}
+
+	s.sendTelemetry(path, startTime, ptest)
+
+	payloadBytes, err := json.Marshal(path)
+	if err != nil {
+		s.logger.Errorf("json marshall error: %s", err)
+	} else {
+		s.logger.Debugf("network path event: %s", string(payloadBytes))
+		m := message.NewMessage(payloadBytes, nil, "", 0)
+		err = s.epForwarder.SendEventPlatformEventBlocking(m, eventplatform.EventTypeNetworkPath)
+		if err != nil {
+			s.logger.Errorf("failed to send event to epForwarder: %s", err)
+		}
+	}
+}
+
+func runTraceroute(cfg traceroute.Config) (payload.NetworkPath, error) {
+	tr, err := traceroute.New(cfg)
+	if err != nil {
+		return payload.NetworkPath{}, fmt.Errorf("new traceroute error: %s", err)
+	}
+	path, err := tr.Run(context.TODO())
+	if err != nil {
+		return payload.NetworkPath{}, fmt.Errorf("run traceroute error: %s", err)
+	}
+	return path, nil
+}
+
+func (s *npCollectorImpl) flushLoop() {
+	s.logger.Debugf("Starting flush loop")
+	defer s.logger.Debugf("Stopped flush loop")
+
+	flushTicker := time.NewTicker(s.flushInterval)
+
+	var lastFlushTime time.Time
+	for {
+		select {
+		// stop sequence
+		case <-s.stopChan:
+			s.logger.Info("Stop flush loop")
+			s.flushLoopDone <- struct{}{}
+			flushTicker.Stop()
+			return
+		// automatic flush sequence
+		case flushTime := <-flushTicker.C:
+			s.flushWrapper(flushTime, lastFlushTime)
+		}
+	}
+}
+
+func (s *npCollectorImpl) flushWrapper(flushTime time.Time, lastFlushTime time.Time) {
+	s.logger.Debugf("Flush loop at %s", flushTime)
+	if !lastFlushTime.IsZero() {
+		flushInterval := flushTime.Sub(lastFlushTime)
+		s.statsdClient.Gauge("datadog.network_path.collector.flush_interval", flushInterval.Seconds(), []string{}, 1) //nolint:errcheck
+	}
+	lastFlushTime = flushTime
+
+	s.flush()
+	s.statsdClient.Gauge("datadog.network_path.collector.flush_duration", s.TimeNowFn().Sub(flushTime).Seconds(), []string{}, 1) //nolint:errcheck
+}
+
+func (s *npCollectorImpl) flush() {
+	s.statsdClient.Gauge("datadog.network_path.collector.workers", float64(s.workers), []string{}, 1) //nolint:errcheck
+
+	flowsContexts := s.pathtestStore.GetContextsCount()
+	s.statsdClient.Gauge("datadog.network_path.collector.pathtest_store_size", float64(flowsContexts), []string{}, 1) //nolint:errcheck
```

Comment:
Do we have the option of using  a distribution here?  It would be great to default size this? I imagine the outliers are going to be way above the p50 here.

Reply:
> Do we have the option of using a distribution here? It would be great to default size this? I imagine the outliers are going to be way above the p50 here.

@leeavital It's indeed possible to send Distribution metrics via DSD. 

Would distribution metric help here? Do you mean metric p50 across orgs?
The metrics we are sending here are sent as custom metrics (for customer org), so not tagged by org_id.